src/layouts/BaseLayout.astro

```diff
@@ -100,30 +101,58 @@ const social_icon = "/Images/mockuphone-logo.png";
   <body>
     <div id="main-page">
       <header class="header">
-        <a
-          href="https://www.skymakers.co.uk/?utm_medium=web&utm_source=mockuphone"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          <Image
-            class="header__skymaker-logo"
-            src="/Images/oursky/skymaker-logo.svg"
-            alt="skymaker-logo.svg"
-            format="svg"
-            width={50}
-            height={48.5}
-          />
-        </a>
-        <a class="header__mockuphone-link" href="/">
-          <Image
-            class="header__mockuphone"
-            src="/Images/oursky/mockuphone-header.svg"
-            alt="mockuphone_header.svg"
-            format="svg"
-            width={50}
-            height={48.5}
-          />
-        </a>
+        <div class="header__logo-group">
+          <a
+            class="header__skymaker-logo-link"
+            href="https://www.skymakers.co.uk/?utm_medium=web&utm_source=mockuphone"
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            <Image
+              class="header__skymaker-logo"
+              src="/Images/oursky/skymaker-logo.svg"
+              alt="skymaker-logo.svg"
+              format="svg"
+              width={50}
+              height={48.5}
+            />
+          </a>
+          <a class="header__mockuphone-link" href="/">
+            <Image
+              class="header__mockuphone"
+              src="/Images/oursky/mockuphone-header.svg"
+              alt="mockuphone_header.svg"
+              format="svg"
+              width={50}
+              height={48.5}
+            />
+          </a>
+        </div>
+
+        {
+          !!shouldRenderSearchBar ? (
+            <div class="header__search-container">
+              <div class="header__search-device__container device-list__search-container">
+                <img src="/Images/search.svg" />
+                <input
+                  class="header__search-device__input device-list__search-input"
+                  placeholder="Search Device"
+                  type="search"
+                />
+                <button
+                  class="header__search-device__clear-btn d-none device-list__search-input__clear-btn"
+                  data-tippy-content="Clear"
+                >
+                  <img src="/Images/cross-gray.svg" />
+                </button>
+              </div>
+              {/* Add suggestion list here */}
+            </div>
+          ) : (
+            <div class="header__search-container"> </div>
+          )
+        }
```

Comment:
Have you tried `<slot />`?

https://docs.astro.build/en/basics/astro-components/#slots

Reply:
hvn't tried yet, gave it a quick read, seems it works like react's children 👍 

here we seem to need [`named slots`](https://docs.astro.build/en/basics/astro-components/#named-slots)

But I think here is the question of abstraction
1.  is `searchBar` part of BaseLayout? OR
2. is `searchBar` only part of deviceList
3. is it just a `slot` in the header, which could be anything

IMO it's 1, so even after knowing `slot`, I will decide to keep current implementation

[Think dynamic HTML is also suggested by astro](https://docs.astro.build/en/basics/astro-syntax/#dynamic-html)

Still, we could wrap the search in another component like below